### PR TITLE
CR-1114812 and CR-1114810: Profiling flag streamlining

### PR DESCRIPTION
**Pull request by jvillarre**
#### Problem solved by the commit
Deprecating confusing profiling options and replacing them with clearer options more aligned to the new profiling architecture's model.

#### Documentation impact (if any)
Newly deprecated switches now issue warnings when used and the changelog has been updated to reflect the switch changes.

### CODE REVIEW

File: src/runtime_src/core/common/config_reader.h [hunk 1]
```diff
@@ -156,6 +149,24 @@ get_data_transfer_trace()
   return value;
 }
 
+inline std::string
+get_data_transfer_trace_dep_message()
+{
+  static bool emitted = false ;
+  if (!emitted) {
```
Comment (stsoe) [inline src/runtime_src/core/common/config_reader.h:156]: This is not thread safe, not sure if it matters, probably doesn't?

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/common/config_reader.h (hunk 1 above)
Comment (jvillarre) [reply src/runtime_src/core/common/config_reader.h:156]: This function is added specifically for a problem we have right now where the OpenCL level device offload and the hardware level device offload are both being loaded and the deprecation message can appear twice.  This happens in a single thread at different times, so thread safety isn't necessary.  Additionally, we are working on delivering a change in 2022.1 where device offload is only loaded at the hardware level once we support device level information on the hardware emulation side without going through the OpenCL level, at which point this function can be removed and the deprecation message moved into the appropriate loaders.

File: src/runtime_src/core/common/config_reader.h [hunk 2]
```diff
@@ -303,6 +305,13 @@ get_opencl_device_counter()
   return value;
 }
 
+inline bool
+get_device_counter()
+{
+  static bool value = detail::get_bool_value("Debug.device_counter", false);
```
Comment (hackwa) [inline src/runtime_src/core/common/config_reader.h:311]: @jvillarre @pgschuey would "device_counters" be more accurate then "device_counter"?

**Comment by dayeh-xilinx** [comment]
retest this please. Fixed boardfarm test issue.

**Comment by dayeh-xilinx** [comment]
retest this please. Failed to download metadata for repo 'epel'

**Review by pgschuey — APPROVED** [review]
Changes look good

**Review by hackwa — APPROVED** [review]

**Review by IshitaGhosh — APPROVED** [review]
Change looks good.

**Pull request merged**